File: cabal/src/CLI/Function/Greeter.hs
module CLI.Function.Greeter (printWelcomeMessage, printGoodbyeMessage) where

import System.IO (hPutStrLn, stderr)

import Goodbye.World (printGoodbye)
import Hello.World (printGreeting)

printWelcomeMessage :: [String] -> IO ()
printWelcomeMessage args = do
    putStrLn $ printGreeting args

printGoodbyeMessage :: [String] -> IO ()
printGoodbyeMessage args = do
    hPutStrLn stderr $ printGoodbye args

File: gradle/packages/cli-function/src/main/kotlin/com/app/cli/function/Greeter.kt
package com.app.cli.function

import com.app.goodbye.world.printGoodbye
import com.app.hello.world.printGreeting

object Greeter {
    @JvmStatic
    fun main(args: Array<String>) {
        println("===================================================")
        printWelcomeMessage(args)
        printGoodbyeMessage(args)
        println("===================================================")
    }

    fun printWelcomeMessage(args: Array<String>) {
        println(printGreeting(args))
    }

    fun printGoodbyeMessage(args: Array<String>) {
        System.err.println(printGoodbye(args))
    }
}
